src/Controller/EnMarche/DonationController.php

```diff
@@ -133,4 +137,44 @@ public function resultAction(Request $request, Donation $donation)
             'retry_url' => $retryUrl,
         ]);
     }
+
+    /**
+     * @Route(
+     *     "/mensuel/cancel",
+     *     defaults={"_enable_campaign_silence"=true},
+     *     name="donation_subscription_cancel"
+     * )
+     * @Method("GET")
+     * // TODO: make test
+     */
+    public function cancelSubscriptionAction(Request $request, DonationRepository $donationRepository, PayboxPaymentUnsubscription $payboxPaymentUnsubscription): RedirectResponse
+    {
+        $donations = $donationRepository->findAllSubscribedDonationByEmail($this->getUser()->getEmailAddress());
```

Comment:
1) why `findAll`? can you have more that one subscription?
2) How one non-adherent user (who can't connect to the platform) can unsubscribe?

Reply:
Yes in database it's possible for BC. There are an  other feature that bloque the user , but before that maybe some users have multiple subscriptions

Reply:
Actualy it's not possible, the feature said that we can unsubscribe by going in Donations page to unsubscribe.

If an anonymous user want unsubscribe we have to create a special flow to confirm by email the fact that he want unsubscribe.